apps/aecore/src/aec_vm_chain.erl

```diff
@@ -663,10 +663,10 @@ call_contract(Target, Gas, Value, CallData, CallStack,
                     GasUsed = aect_call:gas_used(Call),
                     Result  =
                         case aect_call:return_type(Call) of
-                            %% TODO: currently we don't set any
-                            %%       sensible return value on exceptions
                             error ->
-                                aevm_chain_api:call_exception(out_of_gas, GasUsed);
+                                Bin = aect_call:return_value(Call),
+                                ReturnAtom = binary_to_error(Bin),
```

Comment:
I see this is needed otherwise aevm_chain_api:return_value sees the binary and returns ok.